scripts/npm-audit-comparison.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+set -o errexit
+set -o pipefail
+set -o nounset
+set -o xtrace
+
+# Run npm audit on current branch and compare it with the results of running npm audit on the base branch that is set in the package.json. If there are more
+# high or critical findings in the current branch, then the test fails. If the same number of findings are found, then check that the vulnerabilities are the same.
+# If they are are different, then the test fails.
+
+# Save high and critical vulns from current branch
+npm ci
+npm audit | grep -E "(High)" -B3 -A10 > current-branch-high-vulnerabilities.txt || true
+npm audit | grep -E "(Critical)" -B3 -A10 > current-branch-critical-vulnerabilities.txt || true
+
+HIGH_VULN="$(grep -o High current-branch-high-vulnerabilities.txt | wc -l || true)"
+CRITICAL_VULN="$(grep -o Critical current-branch-critical-vulnerabilities.txt | wc -l || true)"
+echo $HIGH_VULN > compare-num-vulnerabilities.txt
+echo $CRITICAL_VULN >> compare-num-vulnerabilities.txt
+
+git stash
+
+# Save high and critical vulns from the base branch (e.g. develop, hotfix/1.11.2)
+git checkout "$npm_package_config_base_branch"
```

Comment:
If you run this locally, won't you:

1. End up with your branch switched to base branch
2. Any local changes stashed (from line 21)

Normally only the CI server will run this, but presumably one day we'll introduce a vulnerability that this will catch, and then we'll want to see what's going on locally, and then it might be confusing.

Maybe just print out a warning message at least? I was going to say always do `git stash pop`, but if `git stash` didn't have anything to save, that will be a mess.

Or maybe don't do the `git stash` at all. I think that should work on CI; and then leave it up to the developers to stash themselves if they run into issues.

Reply:
Yup, you end up in the base branch. Ended up getting rid of the git stash and added a warning at the top that you'll end up on the base branch and that you may want to stash their local changes.